src/components/ContentPreview/Header.js

```diff
@@ -17,85 +19,129 @@ import ContentOpenWith from '../ContentOpenWith/ContentOpenWith';
-                    <ContentOpenWith
-                        className="bcpr-bcow-btn"
-                        fileId={id}
-                        token={token}
-                        {...contentOpenWithProps}
-                    />
-                )}
-                {canDownload && (
-                    <PlainButton
-                        type="button"
-                        className="bcpr-btn"
-                        onClick={onPrint}
-                        title={printMsg}
-                        aria-label={printMsg}
-                    >
-                        <IconPrint color={COLOR_999} width={22} height={22} />
-                    </PlainButton>
-                )}
-                {canDownload && (
-                    <PlainButton
-                        type="button"
-                        className="bcpr-btn"
-                        onClick={onDownload}
-                        title={downloadMsg}
-                        aria-label={downloadMsg}
-                    >
-                        <IconDownload
-                            color={COLOR_999}
-                            width={18}
-                            height={18}
+            <div className="bp-header bp-base-header">
+                <div className="bcpr-name">
+                    {file ? getIcon(24, file) : null}
+                    <span>{name}</span>
+                </div>
+                <div className="bcpr-btns">
+                    {shouldRenderOpenWith && (
+                        <ContentOpenWith
+                            className="bcpr-bcow-btn"
+                            fileId={id}
+                            token={token}
+                            {...contentOpenWithProps}
                         />
-                    </PlainButton>
-                )}
-                {onClose && (
-                    <PlainButton
-                        type="button"
-                        className="bcpr-btn"
-                        onClick={onClose}
-                        title={closeMsg}
-                        aria-label={closeMsg}
-                    >
-                        <IconClose color={COLOR_999} width={24} height={24} />
-                    </PlainButton>
-                )}
+                    )}
+                    {canAnnotate && (
+                        <PlainButton
+                            type="button"
+                            className="bcpr-btn bp-btn-annotate-draw"
+                            onClick={() => {}}
+                            title={drawMsg}
+                            aria-label={drawMsg}
+                        >
+                            <IconDrawAnnotationMode
+                                color={COLOR_999}
+                                width={18}
+                                height={18}
+                            />
+                        </PlainButton>
+                    )}
+                    {canAnnotate && (
```

Comment:
You could batch all of these under one logical canAnnotate &&. Also, how hare configurations for enabling/disabling certain modes reflected here? This looks like an all or nothing check

Reply:
Good point about the different modes for different doc types. The way this is specified, is to add a `bp-is-hidden` class to the buttons and allow Box Annotations itself to show the ones that are valid. I uncovered an issue switching between file types that the different annotate buttons would not get hidden if they didn't apply to the file type. Will update with logic for this